doc/eip-draft/eip-draft_fair_atomic_swap.md

```diff
@@ -0,0 +1,824 @@
+---
+eip: <to be assigned>
+title: Fair Atomic Swap Smart Contract
+author: Runchao Han <runchao.han@monash.edu>, Haoyu Lin <chris.haoyul@gmail.com>, Jiangshan Yu <jiangshan.yu@monash.edu>
+discussions-to: https://github.com/HAOYUatHZ/fair-atomic-swap/issues
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-08-17
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+
+This EIP provides Atomic Swap Smart Contract templates under both Spot scenario and American Call Option scenario, to mitigate the arbitrage risk introduced by market fluctuation.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Atomic Swap enables two parties to atomically exchange their own cryptocurrencies without trusted third parties. There have been investigations on the (un)fairness of the Atomic Swap protocol, and it is found that the Atomic Swap is equivalent to an American Call Option without the premium, and thus unfair to the swap participant. More specifically, in a fluctuating market, the contract initior can take advantage of the unfairness and earn profit from it, without receiving any penalty. What is more, given the timelock setting (24/48 hrs), as [revealed by the community](https://ethereum-magicians.org/t/eip-xxx-fair-atomic-swap-contract/3549), the profit is considerably high, compared with that of conventional financial assets (stocks and fiat currencies). Therefore the initior have the motivation to arbitrage if allowed to, especially when such an arbitrage is risk-free. To address such a problem, one idea is to introduce premium as penalty. However, as described in this EIP, if introducing premium directly, it is also possible that the participant become the arbitrageur. This EIP proposes two fair Atomic Swap protocols, one is for Spots and the other is for American Call Options, addessing all the issues mentioned above.
+
+
+## Motivation
```

Comment:
I guess we can keep this section, or with some minor modifications...

Reply:
Abstract or Motivation?